### PR TITLE
[hail] Teach Expressions to HTML Show

### PR DESCRIPTION
**Pull request by danking**
Summary of Changes
- Preserve MT structure for entry fields
- Harmonize MT and T `show` interface in a backwards compatible way
- Simplify `Expression.show` code slightly
- A comprehensive, colocated set of `show` tests

### CODE REVIEW

File: hail/python/test/hail/expr/test_show.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import hail as hl
+
+setUpModule = startTestHailContext
```
Comment (tpoterba) [inline hail/python/test/hail/expr/test_show.py]: name error? needs to be imported

File: hail/python/hail/table.py [hunk 2]
```diff
@@ -1430,6 +1430,11 @@ def show(self, n=None, width=None, truncate=None, types=True, handler=None):
         handler : Callable[[str], Any]
             Handler function for data string.
         """
+        if n_rows is not None and n is not None:
```
Comment (tpoterba) [inline hail/python/hail/table.py]: I'm a bit confused about this change

Comment (tpoterba) [reply hail/python/hail/table.py]: it's also not typechecked which will create an error.

File: hail/python/hail/expr/expressions/base_expression.py [hunk 3]
```diff
@@ -650,15 +653,17 @@ def _to_table(self, name):
...
+                to_return = source.select_globals()
         assert self.dtype == to_return[name].dtype, f'type mismatch:\n' \
                                                     f'  Actual:    {self.dtype}\n' \
                                                     f'  Should be: {to_return[name].dtype}'
         return to_return
 
-
-    @typecheck_method(n=int, width=int, truncate=nullable(int), types=bool, handler=anyfunc)
-    def show(self, n=10, width=90, truncate=None, types=True, handler=print):
+    @typecheck_method(n=int, width=int, truncate=nullable(int), types=bool, handler=nullable(anyfunc))
```
Comment (tpoterba) [inline hail/python/hail/expr/expressions/base_expression.py]: why this change to handlers?

Comment (tpoterba) [reply hail/python/hail/expr/expressions/base_expression.py]: d'oh, this is the html fix 🤦‍♂

File: hail/python/hail/table.py (hunk 2 above)
Comment (danking) [reply hail/python/hail/table.py]: I did modify the type check annotation.

In `Expression.show` I don't want to have an `if` that calls `show(n=n)` and `show(n=n_rows)`. I unified the two interfaces so that both `Table` and `MatrixTable` call the number of rows to be shown `n_rows` (and Table continues to support `n` for legacy reasons).

File: hail/python/test/hail/expr/test_show.py (hunk 1 above)
Comment (danking) [reply hail/python/test/hail/expr/test_show.py]: fixed

File: hail/python/hail/table.py (hunk 2 above)
Comment (tpoterba) [reply hail/python/hail/table.py]: oh, I see. OK, sure
